### PR TITLE
DNSng utils cont'd

### PR DESCRIPTION
**Pull request by The-Loeki**
### What does this PR do?
Continue adding stuff to the new DNS client stuff

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/pull/39639

### Tests written?
learning 

@cachedout I'm still here, glad you like my code so much, please add this stuff as well so I can deservedly drop the RFC/WIP/nomerge labels ;)
Also note the `service()` function which should now do SRV lookups the best possible way.

### CODE REVIEW

**Comment by The-Loeki** [comment]
So I'm about done with the functional parts of the DNS client for now, I'm quite happy the way things are going.

Next up I'll start refactoring the existing lookup and client functions to start utilizing this code.
And of course test-test-test-test.

One question: can I rely on `__opts__` being available in `utils`? Or how do I get the running configuration?

**Comment by cachedout** [comment]
I like the direction this going. I think we really want some unit testing in here though.

**Review by cachedout — APPROVED** [review]

File: tests/unit/utils/test_dns.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+# Testing
+from tests.support.unit import skipIf, TestCase
+from tests.support.mock import NO_MOCK, NO_MOCK_REASON, MagicMock, patch, call
+
+
+# Debug
+import pprint
+ppr = pprint.PrettyPrinter(indent=2).pprint
+
```
Comment (s0undt3ch) [inline tests/unit/utils/test_dns.py]: The above pretty printer is no longer required right?

File: tests/unit/utils/test_dns.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+# Python
+
+# Salt
+from salt._compat import ipaddress
+from salt.utils.odict import OrderedDict
+from salt.utils.dns import _to_port, _tree, _weighted_order, _data2rec, _data2rec_group
+
+# Testing
+from tests.support.unit import skipIf, TestCase
+from tests.support.mock import NO_MOCK, NO_MOCK_REASON, MagicMock, patch, call
```
Comment (s0undt3ch) [inline tests/unit/utils/test_dns.py]: If there's no mocking going on, these imports should be removed.

**Comment by cachedout** [comment]
@The-Loeki Did you get a chance to look at the above comments from @s0undt3ch ?

**Comment by The-Loeki** [comment]
Hi guys, I did, but those lines might be around for just a little while more while I try to teach myself some proper testing.

How do you guys like the testing setup so far? Am I doing OK or am I far out of the ballpark here?

File: tests/unit/utils/test_dns.py (hunk 1 above)
Comment (The-Loeki) [inline tests/unit/utils/test_dns.py]: correct, it will disappear soon

**Comment by cachedout** [comment]
@The-Loeki I took at look at the tests here and they look great. Please let me know when you think this is ready for final review. Thanks!

**Comment by cachedout** [comment]
@The-Loeki This will also need a rebase because there are now merge conflicts. Thanks.

**Comment by The-Loeki** [comment]
crap... that borked it up again :(

**Pull request closed without merging**